client/src/pages/locations/Form.js

```diff
@@ -138,16 +140,36 @@ const BaseLocationForm = ({ currentUser, edit, title, initialValues }) => {
                   name="location"
                   component={FieldHelper.ReadonlyField}
                   humanValue={
-                    <>
-                      <Coordinate coord={values.lat} />,{" "}
-                      <Coordinate coord={values.lng} />
-                    </>
+                    <GeoLocation
+                      lat={values.lat}
+                      lng={values.lng}
+                      isSubmitting={isSubmitting}
+                      onClearLocation={() => onClearLocation(setFieldValue)}
+                    />
                   }
                 />
+
```

Comment:
I do not see the point to having a row of read-only lat / lon, followed by an editable lat and lon. Why not just have the editor?
Also, it is probably best if the read-only rendering (as in Show) looks similar to the editor - i.e. 2 widgets on the same row.